### ISSUE TITLE
Big SQL query materializing ontology with rdfs:label in mappings file

### ISSUE THREAD

**Issue by fanavarro** [ontop/ontop#183]
Hi, I'm working with an ontology that has 240 classes, 110 object properties and 194 data properties. I've recently added rdfs:label annotations in several mappings in the mapping file and, after that, I'm not able to perform a materialization of triples due to lack of memory error. Before adding the rdfs:label annotation I was able to do this task without problems.

The following screenshot show how I'm inserting labels through the mappings file:
![imagen](https://cloud.githubusercontent.com/assets/11376639/24500261/f12ae6ba-1544-11e7-98a6-8452c93e09a4.png)

I've checked out the log file. Different queries are executed successfully against the database until the query about rdfs:label appears. The final SQL query is about 8000 lines and the server can't manage it, throwing the exception. However, the following sparql query performed through OntopPro returns 9935 results in 0.372 seconds:
```
PREFIX : <http://www.basecia.es/ontologia#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
select distinct * where {
	?s rdfs:label ?o .
}
```

Am I doing any mistake in the mappings file? Is there another way to add rdfs:label in the mappings?

I'm working with the source code of Ontop version 1.18.1-SNAPSHOT, synchronized with the head branch of git for java code testing and Ontop version 1.18.0. embedded in Protege 5.0.0. The problem seems to be in both versions. I'm attaching the console output (only the part concerning to the rdfs:label query), obtained by using 1.18.1-SNAPSHOT.

If you need some extra information, let me know.

Cheers,
Fran.

[salida.txt](https://github.com/ontop/ontop/files/881889/salida.txt)

**Comment by skomlaebri** [comment]
Hi, does the problem appears with any query you execute with rdfs:label, or only during materialisation and the execution of the construct query?
You can try to higher the java [heap size of protege](https://protegewiki.stanford.edu/wiki/Setting_Heap_Size) and see if it is enough to solve your problem.

**Comment by fanavarro** [comment]
Hi @skomlaebri. As always, thanks for your reply.
It only happens during the materialization.
Casually I've just discovered where is the problem and I was going to post the answer right now. The issue is in the used constructor of QuestMaterializer class, which has the following code:

```
/***
 * 
 * 
 * @param model
 * @throws Exception
 */
public QuestMaterializer(OBDAModel model, Ontology onto, Collection<Predicate> predicates, QuestPreferences preferences,
						 boolean doStreamResults) throws Exception {
	this.doStreamResults = doStreamResults;
	this.model = model;
	this.ontology = onto;

    if(predicates != null && !predicates.isEmpty()){
        this.vocabulary = new HashSet<>(predicates);
    } else {
       this.vocabulary = extractVocabulary(model, onto);
    }

	if (this.model.getSources()!= null && this.model.getSources().size() > 1)
		throw new Exception("Cannot materialize with multiple data sources!");


    //start a quest instance
	if (ontology == null) {
		OntologyVocabulary vb = ofac.createVocabulary();
		
		// TODO: use Vocabulary for OBDAModel as well
		
		for (OClass pred : model.getOntologyVocabulary().getClasses()) 
			vb.createClass(pred.getName());				
		
		for (ObjectPropertyExpression prop : model.getOntologyVocabulary().getObjectProperties()) 
			vb.createObjectProperty(prop.getName());

		for (DataPropertyExpression prop : model.getOntologyVocabulary().getDataProperties()) 
			vb.createDataProperty(prop.getName());

		for (AnnotationProperty prop : model.getOntologyVocabulary().getAnnotationProperties())
			vb.createAnnotationProperty(prop.getName());

		ontology = ofac.createOntology(vb);			
	}
	
	
	preferences.setCurrentValueOf(QuestPreferences.ABOX_MODE, QuestConstants.VIRTUAL);

	questInstance = new Quest(ontology, this.model, preferences);
	questInstance.setQueryingAnnotationsInOntology(true);

	questInstance.setupRepository();
}
```
In the end of the constructor, the default quest preferences are read in order to initialize the reasoner. These preferences include the option "org.obda.owlreformulationplatform.queryingAnnotationsInOntology" set as false due to performance reasons. However, after the properties file is read, this option is set as true manually. My problem dissappear when I comment the line "questInstance.setQueryingAnnotationsInOntology(true);".

Maybe I'm missing something but I think this option must be set as false when Quest is materializing triples to the ontology. The annotations are already in the ontology so it would not be necessary that Quest retrieves them through SQL query.

Could this change affect to another parts of the application?

Regards,
Fran.

**Comment by ghxiao** [comment]
Fixed by #189
